File: sample_finder.py
import bs4
import re
import urllib.request

def find_samples(artist, track):
    samples = []
    url = "https://www.whosampled.com/" + artist + "/" + track + "/"
    try:
        with urllib.request.urlopen(url) as f:
            html_dump = f.read().decode('utf-8')
            html = bs4.BeautifulSoup(html_dump, 'html.parser')
            track_stats = [a.get_text() for a in html.find_all(class_='section-header-title')]
            if "samples" in track_stats[0]:
                a = html.find(class_='list bordered-list')
                for sample_html in a.find_all(class_='listEntry sampleEntry'):
                    artist = sample_html.find(class_='trackArtist').a.get_text()
                    track = sample_html.find(class_='trackName playIcon').get_text()
                    samples.append((track, artist))
            else:
                samples.append("No samples found")
    except:
        samples.append("404 not found")
    print(samples)

def sample_finder_console():
    while True:
        artist = re.sub(' ', '-', input("Artist name: "))
        track = re.sub(' ', '-', input("Track name: "))
        find_samples(artist, track)

sample_finder_console()
# find_samples(import_track_page("Die-Antwoord", "Ugly-Boy"))

File: sample_finder.js
window.addEventListener("load", () => {
  let form = document.getElementById('my-form')
  form.addEventListener("submit", evt => {
    evt.preventDefault(); // before the code
    let artist = form.elements[0].value;
    let track = form.elements[1].value;

    getSample(artist, track)
      .then(res => console.log(document.getElementById("sample").innerHTML=res));

  });
});

async function getSample(artist, track) {
  let corsEscape = "https://paye-ton-corse.herokuapp.com/";
  let requestUrl = "https://www.whosampled.com/" + artist.replace(' ', '-') + "/" + track.replace(' ', '-') + "/";
  let url = corsEscape + requestUrl;
  let samples = [];

  try {
    let res = await fetch(url);
    let html = await res.text();

    let parser = new DOMParser();
    let htmlDoc = parser.parseFromString(html, 'text/html');
    let trackStats = htmlDoc.getElementsByClassName('section-header-title');
    if (trackStats[0].innerHTML.includes("Contains samples")){
      let a = htmlDoc.getElementsByClassName('list bordered-list')[0];
      for (let sampleHtml of a.getElementsByClassName('listEntry sampleEntry')) {
        let artist = sampleHtml.getElementsByClassName('trackArtist')[0].firstElementChild.innerHTML;
        let track = sampleHtml.getElementsByClassName('trackName playIcon')[0].innerHTML;
        samples.push([track, artist]);
      }
    } else {
      samples.push('no samples found');
    }
  }
  catch(err){
    console.log(err);
    samples.push('404 not found');
  }
  return samples;
}
